File: src/usagedash/providers/gemini.py
from __future__ import annotations

from usagedash.config import ProviderConfig
from usagedash.models import ProviderName, ProviderSnapshot
from usagedash.providers.base import PartialUsage, ProviderAdapter, merge_usage


class GeminiAdapter(ProviderAdapter):
    name = ProviderName.GEMINI

    def collect(self, cfg: ProviderConfig) -> ProviderSnapshot:
        partial = PartialUsage(messages=["Gemini parser is not implemented in v2; use manual fields"])
        return merge_usage(self.name, partial, cfg)

File: crates/usagedash-core/src/providers/gemini.rs
use crate::config::ProviderConfig;
use crate::models::Provider;
use crate::providers::{PartialUsage, ProviderAdapter, merge_partial_with_manual};
use anyhow::Result;

pub struct GeminiAdapter;

impl ProviderAdapter for GeminiAdapter {
    fn provider(&self) -> Provider {
        Provider::Gemini
    }

    fn collect(&self, cfg: &ProviderConfig) -> Result<crate::models::ProviderStatus> {
        let partial = PartialUsage {
            messages: vec!["gemini adapter is a stub in v1; configure manual values".to_string()],
            ..Default::default()
        };
        Ok(merge_partial_with_manual(self.provider(), Some(partial), &cfg.manual))
    }
}
